jetty-core/jetty-client/src/test/java/org/eclipse/jetty/client/util/SPNEGOAuthenticationTest.java

```diff
@@ -94,26 +124,21 @@ public void dispose() throws Exception
     private void startSPNEGO(Scenario scenario, Handler handler) throws Exception
     {
         server = new Server();
-        server.setSessionIdManager(new DefaultSessionIdManager(server));
-        HashLoginService authorizationService = new HashLoginService(realm, realmPropsPath.toString());
-        ConfigurableSpnegoLoginService loginService = new ConfigurableSpnegoLoginService(realm, AuthorizationService.from(authorizationService, ""));
-        loginService.addBean(authorizationService);
-        loginService.setKeyTabPath(serviceKeyTabPath);
-        loginService.setServiceName(serviceName);
-        loginService.setHostName(serviceHost);
-        server.addBean(loginService);
-
-        ConstraintSecurityHandler securityHandler = new ConstraintSecurityHandler();
-        Constraint constraint = new Constraint();
-        constraint.setAuthenticate(true);
-        constraint.setRoles(new String[]{"**"}); //allow any authenticated user
-        ConstraintMapping mapping = new ConstraintMapping();
-        mapping.setPathSpec("/secure");
-        mapping.setConstraint(constraint);
-        securityHandler.addConstraintMapping(mapping);
-        authenticator = new ConfigurableSpnegoAuthenticator();
+        HashLoginService hashLoginService = new HashLoginService(realm, ResourceFactory.of(server).newResource(realmPropsPath));
+        SPNEGOLoginService spnegoLoginService = new SPNEGOLoginService(realm, AuthorizationService.from(hashLoginService, ""));
+        spnegoLoginService.setKeyTabPath(serviceKeyTabPath);
+        spnegoLoginService.setServiceName(serviceName);
+        spnegoLoginService.setHostName(serviceHost);
+        // Start the authorizationService along with the loginService.
+        spnegoLoginService.addBean(hashLoginService);
```

Comment:
@sbordet Thanks for these updates... but there is something weird here.   To make the test work it needs a `HashLoginService` as a been and a `SpengoLoginService`, which wraps the HSL, set on the `Authenticator`.

It should just need the SLS, set either as a bean or directly on the `Authenticator`.

So what is looking for the HSL as a bean?

Reply:
> So what is looking for the HSL as a bean?

Nothing is looking for it, as the comment says it's just for lifecycle purposes.

As for the need for both the `SPNEGOLoginService` and the `HashLoginService`, IIRC it's due to the fact that SPNEGO does authentication, but has no concept of roles.
So in order to retrieve the roles, another service is needed, and it just so happen that we can use any of the other existing that we already have.

So we can compose SPNEGO with Hash or JDBC, etc.

Reply:
I've no problem with the composition.  But if it wraps another login service, then it should be an auto-bean, so if it starts it then it manages it.

it is confusing to need one login service as a been and the other as a setter.   I'll fix.